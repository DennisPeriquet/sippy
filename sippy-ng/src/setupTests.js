/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import fetch from 'node-fetch'
import path from 'path'
import { setupPolly } from 'setup-polly-jest'
import toDiffableHtml from 'diffable-html'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => toDiffableHtml(wrapper.html()
  .replace(/id="mui-[0-9]*"/g, '')
  .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, ''))

Enzyme.configure({ adapter: new Adapter() })

// To stub out fetch(), we need to use node-fetch and this line:
global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()

// When injecting multiple filters, material table complains.
const originalError = console.error.bind(console.error)

global.console.error = (log) => {
  if (!log.toString().includes('Warning: Failed')) {
    originalError(log.toString())
  }
}

// Set API URL for Sippy
process.env.REACT_APP_API_URL = 'http://localhost:8080'

// Default PollyJS mode is replay, set to record to record
// new API calls.
if (!process.env.POLLY_MODE) {
  process.env.POLLY_MODE = 'replay'
}

export const setupDefaultPolly = () => {
  const context = setupPolly({
    mode: process.env.POLLY_MODE,
    adapters: [require('@pollyjs/adapter-node-http')],
    persister: require('@pollyjs/persister-fs'),
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, '../__recordings__')
      }
    }
  })
  return context
}

export const expectLoadingPage = (wrapper) => expect(wrapper.find('p').contains('Loading...'))

jest.mock('react-chartjs-2', () => ({
  Bar: () => null,
  Chart: () => null,
  ChartComponent: () => null,
  Doughnut: () => null,
  Line: () => null,
  PolarArea: () => null
}))
