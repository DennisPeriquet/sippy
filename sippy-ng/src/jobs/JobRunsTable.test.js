/** @jest-environment setup-polly-jest/jest-environment-node */

import 'jsdom-global/register'
import React from 'react'
import { act, waitFor } from '@testing-library/react'
import { QueryParamProvider } from 'use-query-params'
import { mount } from 'enzyme'
import { setupDefaultPolly, withoutMuiID } from '../setupTests'
import { BrowserRouter } from 'react-router-dom'
import JobRunsTable from './JobRunsTable'

jest.useRealTimers()

describe('JobRunsTable', () => {
  setupDefaultPolly()

  it('should render correctly', async () => {
    Date.now = jest.spyOn(Date, 'now').mockImplementation(() => new Date(1628691480000))
    const fetchSpy = jest.spyOn(global, 'fetch')

    let wrapper
    await act(async () => {
      wrapper = mount(
                <QueryParamProvider>
                    <BrowserRouter>
                        <JobRunsTable release="4.8" />
                    </BrowserRouter>)
                </QueryParamProvider>)
    })

    expect(wrapper.text()).toContain('Fetching data')

    await waitFor(() => {
      wrapper.update()
      expect(wrapper.text()).not.toContain('Fetching data')
    })

    expect(wrapper.text()).toContain('-e2e-')
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
    expect(fetchSpy).toHaveBeenCalledTimes(1)
  })
})
