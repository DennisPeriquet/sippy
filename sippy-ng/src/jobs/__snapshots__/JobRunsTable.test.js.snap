// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobRunsTable should render correctly 1`] = `
"
<br>
<br>
<div>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-S\\">
        S
      </span>
    </span>
    success
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-F\\">
        F
      </span>
    </span>
    failure (e2e)
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-f\\">
        f
      </span>
    </span>
    failure (other tests)
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-U\\">
        U
      </span>
    </span>
    upgrade failure
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-I\\">
        I
      </span>
    </span>
    setup failure (installer)
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-N\\">
        N
      </span>
    </span>
    setup failure (infra)
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-n\\">
        n
      </span>
    </span>
    failure before setup (infra)
  </span>
  <span class=\\"legend-item\\">
    <span class=\\"results results-demo\\">
      <span class=\\"result result-R\\">
        R
      </span>
    </span>
    running
  </span>
</div>
<div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"
     style=\\"margin-top: 20px;\\"
>
  <div class=\\"MuiDataGrid-root MuiDataGrid-root MuiDataGrid-autoHeight\\"
       role=\\"grid\\"
       aria-colcount=\\"5\\"
       aria-rowcount=\\"5788\\"
       aria-multiselectable=\\"false\\"
  >
    <div>
      <div>
        <div class=\\"root\\">
        </div>
      </div>
    </div>
  </div>
</div>
)
"
`;
